topics/assembly/tutorials/assembly-decontamination/tutorial.md

```diff
@@ -0,0 +1,324 @@
+
+
+
+> <agenda-title></agenda-title>
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Get data
+
+For this tutorial, we are using the genome assembly of the Zebra Finch (*aeniopygia guttata*) generated by the [Vertebrate Genome Project](https://vertebrategenomesproject.org/). We downsized the assembly to 50 of the largest scaffold to keep the running time reasonable, and added some contaminants.
+
+This process will work with any assembly in FASTA format. If your initial dataset is compressed, use the tool `Convert compressed file to uncompressed.` first.
+
+
+> <hands-on-title> Data Upload </hands-on-title>
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo]({{ page.zenodo_link }}):
+>
+>    ```
+>    https://zenodo.org/records/13332983/files/Contaminated_assembly.fasta?download=1
+>    ```
+>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
+>
+> 3. Rename the datasets : `Contaminated Assembly`
+> 4. Check that the datatype is `fasta`
+>
+>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="fasta" %}
+>
+>
+{: .hands_on}
+
+# Genome Masking
+
+Genome masking is the technique of hiding the region of the genome that cary low complexity, such as repeats. Masking a genome reduces the noise and improves the efficacy of the decontamination by emphasizing the regions that contribute the most to the classification ({% cite saini2016gene %}, {% cite maskedref-website %}).
+The first step of our analysis is therefore to mask our genome assembly. There is two type of masking, depending on the requirements of the tools we are using.
+
+- **Soft Masking** The masked regions are replaced by lower case letters
+- **Hard Masking** The masked regions are replaced by a wildcard letter:  `N`s for nucleotide sequences, `X`s for proteins
+
+
+> <details-title> Genome masking for everyone </details-title>
+>
+> Repeat masking is important for a lot of analyses like sequence alignments, classification, gene annotation. Repetitive and low complexity regions cause problems for search and clustering algorithms based on patterns. For gene annotation, repeats can cause non-specific gene hits.
+>
+{: .details}
+
+In this tutorial we will use Kraken2 to classify our scaffolds, and DustMasker to mask our genome. Dustmasker uses soft masking, while Kraken2 uses Hard masking, so we will do some file manipulations along the way.
+
+The first step is to ensure the sequences in our assembly are in upper cases.
+
+
+> <hands-on-title> Convert our assembly file to upper case </hands-on-title>
+>
+> 1. {% tool [Text transformation](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `Contaminated_assembly`
+>    - *"SED Program"*: `s/.*/\U&/` (Convert all letters to uppercase)
```

Comment:
1. Depending on if you think it is useful, you can include our [faq about regular expressions](https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_regular_expressions.html) somewhere around this stip?

2.. Maybe you would like to add this comand for converting to uppercase (and then perhaps also lowercase?) to this FAQ?

Reply:
This will affect not only the lines with the sequences, but also the read IDs etc right? can that have any consequences downstream (in general)? can we adjust the regex to ignore lines that start with `>` or just add a word of warning?

Reply:
I have a question about adding the Lower/upper case transformation to the FAQ. The \l and \U tokens seem to work only on Galaxy (at least for me), the sed in my terminal doesn't understand it, the regex checkers don't know it, yet it's everywhere in forums/articles. Should I still add it with the caveat that it may not work everywhere?

Reply:
@Delphine-L make sure you are using GNU sed. Not the OSX sed. You can get the GNU sed via conda for example.